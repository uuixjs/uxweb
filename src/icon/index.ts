export { Icon, IconProps } from "./component";
