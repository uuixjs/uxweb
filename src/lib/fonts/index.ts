// export * from "./sc-font-declaration";
