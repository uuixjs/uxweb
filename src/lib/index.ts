// export * from './fonts';
export * from './pure-color';
// export * from './test-utils';
// export * from './ui-scss-compat';
export * from './ui-tokens';
export * from './ui-utils';
export * from './utils-ts';
export * from './utils-styled-components';
