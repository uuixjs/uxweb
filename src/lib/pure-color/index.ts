import convert from "./convert";
import parse from "./parse";

export * from "./types";
export * from "./convert";
export * from "./parse";

export const pureColor = {
  convert,
  parse,
};

export default pureColor;
