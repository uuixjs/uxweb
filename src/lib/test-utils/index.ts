export function createMountWrapperFactory(x: any): any {
    return x;
}

export default createMountWrapperFactory;