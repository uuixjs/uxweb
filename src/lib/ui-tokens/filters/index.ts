import { Property } from "../types";

export const excludeThemesCustomFilter = {
  name: "excludeThemes",
  matcher: (prop: Property) => prop.attributes.category !== "themes",
};

export const isThemeColor = {
  name: "isThemeColor",
  matcher: (prop: Property) =>
    prop.attributes.category === "themes" &&
    prop.attributes.item &&
    prop.attributes.item.startsWith("color"),
};
