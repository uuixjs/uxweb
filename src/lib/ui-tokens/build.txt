// @ts-check
import {
  colorUIColorSwiftCustomTransform,
  cssCustomTransformGroup,
  customColorCss,
  iosSwiftSeparateCustomTransformGroup,
  nameTiSnakeCustomTransform,
  scssCustomTransformGroup,
} from "./transforms";
import {
  cssThemesCustomFormat,
  iosSwiftColorsCustomFormat,
  iosSwiftThemesCustomFormat,
  iosThemeProtocolCustomFormat,
  jsonAllCustomFormat,
  jsonThemeCustomFormat,
  jsonVariablesCustomFormat,
  scssColorsCustomFormat,
  scssThemesCustomFormat,
  tsCustomFormat,
  tsDictionaryCustomFormat,
} from "./formats";
import { excludeThemesCustomFilter, isThemeColor } from "./filters";

import StyleDictionary from "style-dictionary";
import { styleDictionaryConfig } from "./style-dictionary.config";

export const buildTokens = () => {
  const styleDictionary = StyleDictionary.extend(styleDictionaryConfig);

  /**
   * A special file configuration is filter, which will filter the tokens
   * before they get to the format. This allows you to re-use the same format
   * to generate multiple files with different sets of tokens.
   *
   * https://amzn.github.io/style-dictionary/#/formats?id=filtering-tokens
   */
  styleDictionary.registerFilter(excludeThemesCustomFilter);
  styleDictionary.registerFilter(isThemeColor);

  /**
   * Transforms are functions that transform a property - this enables each
   * platform to consume the property in different ways.
   *
   * https://amzn.github.io/style-dictionary/#/transforms
   */
  styleDictionary.registerTransform(nameTiSnakeCustomTransform);
  styleDictionary.registerTransform(colorUIColorSwiftCustomTransform);
  styleDictionary.registerTransform(customColorCss);

  /**
   * Transform Groups are a way to easily use multiple transforms at once. They
   * are an array of transforms.
   *
   * https://amzn.github.io/style-dictionary/#/transform_groups
   */
  styleDictionary.registerTransformGroup(cssCustomTransformGroup);
  styleDictionary.registerTransformGroup(scssCustomTransformGroup);
  styleDictionary.registerTransformGroup(iosSwiftSeparateCustomTransformGroup);

  /**
   * Formats define the output of your created files.
   *
   * https://amzn.github.io/style-dictionary/#/formats
   */
  styleDictionary.registerFormat(cssThemesCustomFormat);
  styleDictionary.registerFormat(scssThemesCustomFormat);
  styleDictionary.registerFormat(scssColorsCustomFormat);
  styleDictionary.registerFormat(iosSwiftColorsCustomFormat);
  styleDictionary.registerFormat(iosThemeProtocolCustomFormat);
  styleDictionary.registerFormat(iosSwiftThemesCustomFormat);
  styleDictionary.registerFormat(jsonAllCustomFormat);
  styleDictionary.registerFormat(jsonThemeCustomFormat);
  styleDictionary.registerFormat(jsonVariablesCustomFormat);
  styleDictionary.registerFormat(tsCustomFormat);
  styleDictionary.registerFormat(tsDictionaryCustomFormat);

  // Run the build command.
  styleDictionary.buildAllPlatforms();
};
