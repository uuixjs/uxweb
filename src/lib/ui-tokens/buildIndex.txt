import { buildTokens } from "./build";

buildTokens();
