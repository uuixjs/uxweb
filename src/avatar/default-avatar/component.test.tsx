import {
  DefaultAvatarComponent,
  DefaultAvatarProps,
  ScDefaultAvatar,
} from "./component";

import { BorderRadius } from "lib";
import { Icon } from "../../icon";
import { StaticTokenMap } from "lib/ui-tokens";
import { setupShallowTest } from "../../../tests/helpers";

function getRequiredProps(): DefaultAvatarProps {
  return {
    userLogin: "pondelinip",
    size: 30,
  };
}

function getOptionalProps(): DefaultAvatarProps {
  return {
    ...getRequiredProps(),
    borderRadius: BorderRadius.Medium,
  };
}

const setupShallow = setupShallowTest(DefaultAvatarComponent, getRequiredProps);

describe("DefaultAvatar", () => {
  it("renders default-avatar component with required props", () => {
    const { wrapper } = setupShallow();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders default-avatar component with all props", () => {
    const { wrapper } = setupShallow(getOptionalProps());
    expect(wrapper).toMatchSnapshot();
  });

  it("skips the icon for small sizes", () => {
    const { wrapper } = setupShallow({ size: 10 });
    expect(wrapper.find(Icon)).not.toExist();

    const { wrapper: anotherWrapper } = setupShallow({ size: 15 });
    expect(anotherWrapper.find(Icon)).not.toExist();
  });

  it.each([
    // The qoi0 username was arbitrarily chosen to match accent-carrot, to make it easier to keep
    // this test and the DEFAULT_AVATAR_COLORS array in sync (they can both be in alphabetical order)
    [StaticTokenMap["color-brand-accent-carrot"], "qoi0"],
    [StaticTokenMap["color-brand-accent-dragonfruit"], "qoi1"],
    [StaticTokenMap["color-brand-accent-eggplant"], "qoi2"],
    [StaticTokenMap["color-brand-accent-grape"], "qoi3"],
    [StaticTokenMap["color-brand-accent-lime"], "qoi4"],
    [StaticTokenMap["color-brand-accent-marine"], "qoi5"],
    [StaticTokenMap["color-brand-accent-seafoam"], "qoi6"],
    [StaticTokenMap["color-brand-accent-sun"], "qoi7"],
    [StaticTokenMap["color-brand-accent-turquoise"], "qoi8"],
    [StaticTokenMap["color-brand-accent-wine"], "qoi9"],
    [StaticTokenMap["color-brand-muted-emerald"], "qoi10"],
    [StaticTokenMap["color-brand-muted-ice"], "qoi11"],
    [StaticTokenMap["color-brand-muted-mustard"], "qoi12"],
    // This accent-carrot is repeated to demonstrate that the assignment 'loops around'
    [StaticTokenMap["color-brand-accent-carrot"], "qoi13"],
  ])('assigns color "%s", given user login "%s"', (color, userLogin) => {
    const { wrapper } = setupShallow({ userLogin });

    expect(wrapper.find(ScDefaultAvatar)).toHaveStyleRule(
      "background-color",
      color,
    );
  });
});
