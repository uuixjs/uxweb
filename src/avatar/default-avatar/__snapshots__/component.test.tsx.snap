// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultAvatar renders default-avatar component with all props 1`] = `
<styled.div
  borderRadius="border-radius-medium"
  userLogin="pondelinip"
>
  <styled.div
    className="tw-default-avatar__icon"
  >
    <Icon
      asset="Account"
      fill={true}
    />
  </styled.div>
</styled.div>
`;

exports[`DefaultAvatar renders default-avatar component with required props 1`] = `
<styled.div
  userLogin="pondelinip"
>
  <styled.div
    className="tw-default-avatar__icon"
  >
    <Icon
      asset="Account"
      fill={true}
    />
  </styled.div>
</styled.div>
`;
